### DISCUSSION TITLE
Appending point to 2D plot

### DISCUSSION THREAD

**Discussion by IasonManolas** [alandefreitas/matplotplusplus#100] [Q&A]
I want to iteratively add points to a 2d plot similar to [this](https://stackoverflow.com/questions/11874767/how-do-i-plot-in-real-time-in-a-while-loop-using-matplotlib#comment58108667_15720891) post.
Can I do this with matplotplusplus?

**Comment by alandefreitas** [comment]
If you want to iteratively *add* points to a plot, have a look at this: https://alandefreitas.github.io/matplotplusplus/examples/appearance/multiplot/hold.html

If you want to iteratively *change* the points in a plot, just put the commands in a loop without `hold`.

**Reply by IasonManolas** [reply]
Isn't hold in these examples used to add lines and not _points_ to the same plot? 
I expected something along the lines of:
plotHandle->addPoint(p_x,p_y);

**Reply by alandefreitas** [reply]
"Hold" means the next plot (be it points, lines, histograms, bar plots, contour lines, whatever) will not replace the previous plot.
